tests/e2e/gc_rwx_multi_ns_gc.go

```diff
@@ -0,0 +1,293 @@
+		26. Verify if the Pod is successfully deleted
+		27. Delete PVC in GC
+		28. Verify if PVC is deleted in GC
+		29. Delete PVs
+		30. Verify if PVs are deleted in the SV cluster and GC
+		31. Verify CnsVolumeMetadata CRD is deleted
+		32. Check if the VolumeID is deleted from CNS by using CNSQuery API
+	*/
+	ginkgo.It("[rwm-csi-tkg] Verify static volume provisioning works across Namespace in the same GC", func() {
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+
+		var storageclasspvc *storagev1.StorageClass
+		var pvclaim *v1.PersistentVolumeClaim
+		var pv *v1.PersistentVolume
+		var err error
+
+		ginkgo.By("CNS_TEST: Running for GC setup")
+		ginkgo.By("Creating second namespace on GC")
+		ns, err := framework.CreateTestingNS(f.BaseName, client, map[string]string{
+			"e2e-framework": f.BaseName,
+		})
+		gomega.Expect(err).NotTo(gomega.HaveOccurred(), "Error creating namespace on second GC")
+
+		newnamespace := ns.Name
+		framework.Logf("Created second namespace on GC %v", newnamespace)
+		defer func() {
+			err := client.CoreV1().Namespaces().Delete(ctx, newnamespace, *metav1.NewDeleteOptions(0))
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}()
+
+		scParameters[svStorageClassName] = storagePolicyName
+
+		ginkgo.By("Creating a PVC")
+		storageclasspvc, pvclaim, err = createPVCAndStorageClass(client,
+			namespace, nil, scParameters, diskSize, nil, "", false, v1.ReadWriteMany)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		defer func() {
+			err = client.StorageV1().StorageClasses().Delete(ctx, storageclasspvc.Name, *metav1.NewDeleteOptions(0))
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}()
+
+		ginkgo.By("Expect claim to provision volume successfully")
+		persistentvolumes, err := fpv.WaitForPVClaimBoundPhase(client,
+			[]*v1.PersistentVolumeClaim{pvclaim}, framework.ClaimProvisionTimeout)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred(), "Failed to provision volume")
+
+		pvcNameInSV := persistentvolumes[0].Spec.CSI.VolumeHandle
+		gomega.Expect(pvcNameInSV).NotTo(gomega.BeEmpty())
+		fcdIDInCNS := getVolumeIDFromSupervisorCluster(pvcNameInSV)
+		gomega.Expect(fcdIDInCNS).NotTo(gomega.BeEmpty())
+
+		defer func() {
+			if pvclaim != nil {
+				err = fpv.DeletePersistentVolumeClaim(client, pvclaim.Name, pvclaim.Namespace)
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+				err = e2eVSphere.waitForCNSVolumeToBeDeleted(fcdIDInCNS)
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			}
+		}()
+
+		// Verify using CNS Query API if VolumeID retrieved from PV is present.
+		framework.Logf("Invoking QueryCNSVolumeWithResult with VolumeID: %s", fcdIDInCNS)
+		queryResult, err := e2eVSphere.queryCNSVolumeWithResult(fcdIDInCNS)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		gomega.Expect(queryResult.Volumes).ShouldNot(gomega.BeEmpty())
+		framework.Logf("volume Name:%s, capacity:%d volumeType:%s health:%s accesspoint: %s",
```

Comment:
where are we verifying the cns entry

Reply:
gomega.Expect(queryResult.Volumes).ShouldNot(gomega.BeEmpty())

Reply:
queryCNSVolumeWithResult function checks the entry in CNS

Reply:
we need to verify all fields

Reply:
Added validations for volume's capacity check and volumetype check

Reply:
look at `verifyVolumeMetadataInCNS()`

Reply:
Line 203 is added for the same
		//Add a check to validate CnsVolumeMetadata crd
		err = waitAndVerifyCnsVolumeMetadata4GCVol(fcdIDInCNS, pvcNameInSV, pvclaim, pv, nil)
		gomega.Expect(err).NotTo(gomega.HaveOccurred())